### PR TITLE
Memory functions

### PR DESCRIPTION
**Pull request by Mograbi**
added memory functionality to basic UI.
I think it will benefit everyone.
did some tests manually, didn't know how to add tests .

### CODE REVIEW

**Comment by jeremypw** [comment]
@Mograbi  It would be good if you could address the changes requested and the conflicts that have now arisen with the master branch so that this PR can be merged :-)

File: src/MainWindow.vala [hunk 1]
```diff
@@ -268,6 +306,10 @@ namespace PantheonCalculator {
             button_9.clicked.connect (() => {regular_button_clicked (button_9.function);});
             button_point.clicked.connect (() => {regular_button_clicked (button_point.function);});
             button_percent.clicked.connect (() => {regular_button_clicked (button_percent.function);});
+            button_mem_clr.clicked.connect (() => {button_mem_clr_clicked ();});
```
Comment (donadigo) [inline src/MainWindow.vala]:
```suggestion
            button_mem_clr.clicked.connect (button_mem_clr_clicked);
```

File: src/MainWindow.vala [hunk 2]
```diff
@@ -268,6 +306,10 @@ namespace PantheonCalculator {
...
+            button_mem_recall.clicked.connect (() => {button_mem_recall_clicked ();});
```
Comment (donadigo) [inline src/MainWindow.vala]:
```suggestion
            button_mem_recall.clicked.connect (button_mem_recall_clicked);
```

File: src/MainWindow.vala [hunk 3]
```diff
@@ -268,6 +306,10 @@ namespace PantheonCalculator {
...
+            button_mem_add.clicked.connect (() => {button_mem_add_clicked ();});
```
Comment (donadigo) [inline src/MainWindow.vala]:
```suggestion
            button_mem_add.clicked.connect (button_mem_add_clicked);
```

File: src/MainWindow.vala [hunk 4]
```diff
@@ -268,6 +306,10 @@ namespace PantheonCalculator {
...
+            button_mem_sub.clicked.connect (() => {button_mem_sub_clicked ();});
```
Comment (donadigo) [inline src/MainWindow.vala]:
```suggestion
            button_mem_sub.clicked.connect (button_mem_sub_clicked);
```

File: src/MainWindow.vala [hunk 5]
```diff
@@ -394,6 +436,54 @@ namespace PantheonCalculator {
             entry.set_position (position);
         }
 
+        private void button_mem_clr_clicked () {
+            memory = null;
+
+            button_mem_recall.sensitive = false;
+            button_mem_clr.sensitive = false;
```
Comment (donadigo) [inline src/MainWindow.vala]: Setting `memory` to `null` already makes these buttons insensitive.

File: src/MainWindow.vala [hunk 6]
```diff
@@ -394,6 +436,54 @@ namespace PantheonCalculator {
...
+        private void button_mem_recall_clicked () {
+            entry.set_text (memory);
+        }
+
+        private void memory_function (string functionality) {
+            var stored_mem = memory;
+            if (memory == null) {
+                stored_mem = "0";
+            }
+            var result = stored_mem + functionality + entry.get_text ();
```
Comment (donadigo) [inline src/MainWindow.vala]: Newline after `}`.

File: src/MainWindow.vala [hunk 7]
```diff
@@ -394,6 +436,54 @@ namespace PantheonCalculator {
...
+
+            memory = result;
+        }
+
+        private void button_mem_add_clicked () {
+            if (entry.get_text () != "") {
+                memory_function ("+");
+
+                button_mem_recall.sensitive = true;
+                button_mem_clr.sensitive = true;
```
Comment (donadigo) [inline src/MainWindow.vala]: These are not needed, they will be set anyway indirectly by the `memory_function` call.

File: src/MainWindow.vala [hunk 8]
```diff
@@ -394,6 +436,54 @@ namespace PantheonCalculator {
...
+            }
+        }
+
+        private void button_mem_sub_clicked () {
+            if (entry.get_text () != "") {
+                memory_function ("-");
+
+                button_mem_recall.sensitive = true;
+                button_mem_clr.sensitive = true;
```
Comment (donadigo) [inline src/MainWindow.vala]: Same here.

**Comment by jeremypw** [comment]
@Mograbi May I assume you are no longer interested in working on this PR, in which case I may fork it?

**Comment by Mograbi** [comment]
@jeremypw yes you can
Thanks

**Comment by jeremypw** [comment]
Closed in favour of #141

**Pull request closed without merging**